iroh-ctl/src/gateway.rs

```diff
@@ -0,0 +1,7 @@
+#[derive(Args, Debug, Clone)]
+pub struct Gateway {
+    #[clap(subcommand)]
+    command: GatewayCommands
```

Comment:
For the GatewayCommands, I'd generally keep this very lean. The gateway does 1 thing only and that's handle `GET` requests. 

I'd add a `Get` cmd which would be the equivalent of running curl against the gateway and just pipe the output to shell directly. User is responsible for writing it wherever they like. It's useful for testing purposes especially as a `ctl` interface. An addition to this would also be a `Head` command, which would just do a `HEAD` request (we don't really support this yet but should) which would just return headers, that way we have a simple interface to get the body and the metadata. 

Everything else on the gateway is pure config stuff and would leave that out of this.

Reply:
stubbed in `get` and `head` commands